File: src/utils/excerpt.js
function stripMarkdown(markdown = '') {
    return markdown
        .replace(/```[\s\S]*?```/g, '')
        .replace(/`([^`]*)`/g, '$1')
        .replace(/\!\[[^\]]*\]\([^)]*\)/g, '')
        .replace(/\[([^\]]*)\]\([^)]*\)/g, '$1')
        .replace(/[#>*_~\-]/g, '')
        .replace(/\s+/g, ' ')
        .trim();
}

function buildExcerpt(markdown = '', maxWords = 40) {
    const plain = stripMarkdown(markdown);
    if (!plain) return '';
    const words = plain.split(' ');
    if (words.length <= maxWords) {
        return plain;
    }
    return `${words.slice(0, maxWords).join(' ')}…`;
}

module.exports = {
    buildExcerpt,
};

File: excerpt.py
import re

_STRIP = [
    (re.compile(r"```[\s\S]*?```"), ""),
    (re.compile(r"`([^`]*)`"), r"\1"),
    (re.compile(r"!\[[^\]]*\]\([^)]*\)"), ""),
    (re.compile(r"\[([^\]]*)\]\([^)]*\)"), r"\1"),
    (re.compile(r"[#>*_~\-]"), ""),
    (re.compile(r"\s+"), " "),
]


def build_excerpt(markdown: str, max_words: int = 40) -> str:
    text = markdown
    for pattern, repl in _STRIP:
        text = pattern.sub(repl, text)
    text = text.strip()
    if not text:
        return ""
    words = text.split()
    return " ".join(words[:max_words]) + ("…" if len(words) > max_words else "")
